probability/exponential_dist.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file
+ * @brief [Exponential
+ * Distribution](https://en.wikipedia.org/wiki/Exponential_distribution)
+ *
+ * The exponential distribution is used to model
+ * events occuring between a Poisson process like radioactive decay.
+ *
+ * \f[P(x, \lambda) = \lambda e^{-\lambda x}\f]
+ *
+ * Summary of variables used:
+ * \f$\lambda\f$ : rate parameter
+ */
+
+#include <cassert>   // For asserting the test cases
+#include <cmath>     // For power function
+#include <iostream>  // For I/O operation
+
+/**
+ * @brief the expected value of the exponential distribution
+ * @returns \f[\mu = \frac{1}{\lambda}\f]
+ */
+double exponential_expected(double lambda) {
+    if (lambda <= 0) {
+        std::cout << "Error: Lambda must be greater than 0." << '\n';
+        assert(lambda > 0);
+    }
+    return 1 / lambda;
+}
+
+/**
+ * @brief the variance of the exponential distribution
+ * @returns \f[\sigma^2 = \frac{1}{\lambda^2}\f]
+ */
+double exponential_var(double lambda) {
+    if (lambda <= 0) {
+        std::cout << "Error: Lambda must be greater than 0." << '\n';
+        assert(lambda > 0);
+    }
+    return 1 / pow(lambda, 2);
+}
+
+/**
+ * @brief the standard deviation of the exponential distribution
+ * @returns \f[\sigma = \frac{1}{\lambda}\f]
+ */
+double exponential_std(double lambda) {
+    if (lambda <= 0) {
+        std::cout << "Error: Lambda must be greater than 0." << '\n';
+        assert(lambda > 0);
+    }
+    return 1 / lambda;
+}
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void test() {
+    double lambda_1 = 1;
+    double expected_1 = 1;
+    double var_1 = 1;
+    double std_1 = 1;
+
+    double lambda_2 = 2;
+    double expected_2 = 0.5;
+    double var_2 = 0.25;
+    double std_2 = 0.5;
+
+    double lambda_3 = 3;
+    double expected_3 = 0.333333;
+    double var_3 = 0.111111;
+    double std_3 = 0.333333;
+
```

Comment:
add some tests with negative numbers and zero

Reply:
The value of lambda is always greater than 0 that's why I haven't added them. It is in assertion in the functions.

Reply:
your algorithm should throw a runtime error in such case, and you should account for this by asserting that the error is indeed thrown. in a short example it could be implemented as follows
```cpp
int my_algorithm(double my_val) {
   if (my_val < 0) {
      throw std::invalid_argument("my_val cant be negative");
   }
}


...

static void tests() {
   try {
      my_algorithm(-3.14);
   } except(std::invalid_argument& err) {
      assert(err.what() == "my_val cant be negative");
   }
}



```

Reply:
@realstealthninja  I have added tests for 0 and negative values